nadabot/models.py

```diff
@@ -0,0 +1,187 @@
+        _("admin notes"),
+        blank=True,
+        null=True,
+        help_text=_("Admin notes, e.g. reason for flagging or marking inactive.")
+    )
+    default_weight = models.PositiveIntegerField(
+        _("default weight"),
+        null=False,
+        help_text=_("Default weight for this provider, e.g. 100.")
+    )
+    gas = models.BigIntegerField(
+        _("gas"),
+        blank=True,
+        null=True,
+        help_text=_("Custom gas amount required.")
+    )
+    tags = models.JSONField(
+        _("tags"),
+        blank=True,
+        null=True,
+        help_text=_("Optional tags.")
+    )
+    icon_url = models.URLField(
+        _("icon URL"),
+        blank=True,
+        null=True,
+        help_text=_("Optional icon URL.")
+    )
+    external_url = models.URLField(
+        _("external URL"),
+        blank=True,
+        null=True,
+        help_text=_("Optional external URL.")
+    )
+    submitted_by = models.CharField(
+        _("submitted by"),
+        max_length=100,
+        null=False,
+        help_text=_("User who submitted this provider.")
+    )
+    submitted_at_ms = models.DateTimeField(
+        _("submitted at (milliseconds)"),
+        null=False,
+        help_text=_("Timestamp of when this provider was submitted.")
+    )
+    stamp_count = models.BigIntegerField(
+        _("stamp count"),
+        default=0,
+        help_text=_("Total number of times this provider has been used successfully."),
+        db_index=True
+    )
+    stamp_validity_ms = models.BigIntegerField(
+        _("stamp validity"),
+        blank=True,
+        null=True,
+        help_text=_("Milliseconds that stamps from this provider are valid for before they expire.")
+    )
+    account_id_arg_name = models.CharField(
+        _("account ID argument name"),
+        max_length=100,
+        null=False,
+        help_text=_("Name of account ID argument, e.g. 'account_id' or 'accountId' or 'account'.")
+    )
+    custom_args = models.CharField(
+        _("custom args"),
+        null=True,
+        help_text=_("Custom args as Base64VecU8.")
+    )
+    registry = models.ForeignKey(
+        NadabotRegistry,
+        on_delete=models.CASCADE,
+        verbose_name=_("registry"),
+        help_text=_("registry under which provider was registered")
+    )
+
+    class Meta:
+        indexes = [models.Index(fields=["id", "status"], name="idx_provider_id_status")]
+
+
+class Stamp(models.Model):
+    user = models.ForeignKey(
+        Account,
+        on_delete=models.CASCADE,
+        verbose_name=_("user"),
+        help_text=_("The user who earned the stamp.")
+    )
+    provider = models.ForeignKey(
+        Provider,
+        on_delete=models.CASCADE,
+        verbose_name=_("provider"),
+        help_text=_("The provider the user verified with.")
+    )
+    verification_date = models.DateField(
+        _("verification date"),
+        auto_now_add=True,
+        help_text=_("The date of verification.")
+    )
```

Comment:
@Prometheo can you update these models to implement nadabot v2? e.g. include `custom_args` field for `Provider`, add `Group` model

Reply:
is nadabot v2 already deployed?

Reply:
There's some more info in this ticket: https://github.com/PotLock/nadabot-app/issues/109

Lmk if that clarifies or if you need more info